### PR TITLE
[Enhancement] Optimize report by eliminate duplicate report

### LINKED ISSUE #6466: Optimize report request queue strategic
## Enhancement
since BE report request is idempotent, so use the latest request to replace the former one, it will reduce FE pressure when report request backlog.

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6466

## Problem Summary(Required) ：

since BE report request is idempotent, so that use the latest request to replace the former one, reduce FE pressure.

### CODE REVIEW

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Review by trueeyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java [hunk 1]
```diff
@@ -100,10 +100,20 @@
 import java.util.concurrent.BlockingQueue;
 
 public class ReportHandler extends Daemon {
+    public enum ReportType {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java:103]:
```suggestion
    public enum ReportType {
        UNKNOWN,
        TABLET,
        DISK,
        TASK,
        WORKGROUP
    }
```

File: fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java [hunk 2]
```diff
@@ -113,6 +123,10 @@ public Long getValue() {
             }
         };
         MetricRepo.addMetric(gaugeQueueSize);
+        pendingTaskMap.put(ReportType.TABLET_REPORT, Maps.newHashMap());
+        pendingTaskMap.put(ReportType.DISK_REPORT, Maps.newHashMap());
+        pendingTaskMap.put(ReportType.TASK_REPORT, Maps.newHashMap());
+        pendingTaskMap.put(ReportType.WORKGROUP_REPORT, Maps.newHashMap());
     }
 
     public TMasterResult handleReport(TReportRequest request) throws TException {
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java:132]: Is it possible that the request contains reports of both task & disk? The original code seems to handle this situation properly, yet after this PR every ReportTask is deduplicated by its last ReportType.

File: fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java [hunk 3]
```diff
@@ -191,14 +208,18 @@ public TMasterResult handleReport(TReportRequest request) throws TException {
     }
 
     private void putToQueue(ReportTask reportTask) throws Exception {
-        int currentSize = reportQueue.size();
-        if (currentSize > Config.report_queue_size) {
```
Comment (padmejin) [inline fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java:195]: Better add a deprecated modifier to this config.

File: fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java (hunk 2 above)
Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java:132]: It's not possible since it has checked BE code

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 53 (20.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|11|53|20.75%|

**Review by chaoyli — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-2.2

**Comment by meegoo** [comment]
@mergify backport branch-2.1

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 53 (20.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|11|53|20.75%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 53 (20.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|11|53|20.75%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 53 (20.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|11|53|20.75%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 53 (20.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|11|53|20.75%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 11 / 53 (20.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|11|53|20.75%|

**Pull request merged**